src/pytest_replay/__init__.py

```diff
@@ -39,6 +44,24 @@ def pytest_addoption(parser):
     )
 
 
+@dataclasses.dataclass
+class ReplayTestMetadata:
+    nodeid: str
+    start: float = 0.0
+    finish: Optional[float] = None
+    outcome: Optional[str] = None
+    metadata: dict[str, Any] = dataclasses.field(default_factory=dict)
+
+    def to_clean_dict(self) -> dict[str, Any]:
+        return {k: v for k, v in asdict(self).items() if v}
+
+
+class _ReplayTestMetadataDefaultDict(collections.defaultdict):
```

Comment:
As per the other comment, for consistency:

```suggestion
class _ReplayTestInfoDefaultDict(collections.defaultdict):
```